zk/arkworks/gramine/src/settlement/proof.rs

```diff
@@ -466,15 +474,18 @@ fn check_satisfaction(
 
     let mut expected_output_notes = vec![];
     for note in private.input_notes.iter().take(unpadded_len) {
-        let note = {
-            let remainder = note.amount() - private.setoff_amount;
-            let new_value = Value {
-                amount: remainder,
-                asset_id: note.asset_id(),
+        // TODO: zero-valued input notes are not allowed right
```

Comment:
We calculate `unpadded_len` based on the position of the first non-zero input note. Ig there's an implicit assumption that the real notes will have non-zero amounts and will be the initial/starting notes.